source/Ship.h

```diff
@@ -170,6 +170,10 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	void SetPosition(Point position);
 	// When creating a new ship, you must set the following:
 	void Place(Point position = Point(), Point velocity = Point(), Angle angle = Angle(), bool isDeparting = true);
+	double turretRange = 0;
+	double gunRange = 0;
```

Comment:
these should be in the private section of the class

Reply:
Why? I understand that it is best practices for data security to have the variable be private and then have encapsulation to extricate the info inside the pirvate variable; however in practice that just seem to add extra steps for no obvious benefit.

That being said, I'm trying to change this, and having trouble getting it to successfully build. Moving those variables into a private area is easy enough, but I am failing to figure out how to encapsulate it properly.

Beyond that, this looks like the same sort of situation as the antiMissileRange value, which is likewise a public variable.

Reply:
Making these fields (here, a `field` is a data member, not a method/function) public means anything with non-const access to a `Ship` object can modify them. This is undesirable. `antiMissileRange` is definitely not public.

Reply:
OK, I'll change this. It is, however, going to take a couple consecutive days in which I get a good amount of sleep in order to have my brain working well enough to figure this out. Just stating that here so you don't think I'm ignoring you or anything. I'm just not currently in a brain space where I can figure out exactly how to do this. Pointers & suggestions appreciated.

Reply:
OK, this has been moved. Thanks for pointing it out. And thanks to Hurleveur for helping!